Makefile

```diff
@@ -33,18 +62,37 @@ publish-on-pypi: venv
 		echo Working directory is dirty >&2; \
 	fi;
 
+.PHONY: regenerate-test-figures
+regenerate-test-figures: $(VENV_DIR)  ## re-generate all test figures
+	$(VENV_DIR)/bin/pytest --mpl-generate-path=tests/expected_figs tests/test_plotting.py
+
 .PHONY: test
-test: venv
-	./venv/bin/pytest tests
+test: $(VENV_DIR)  ## run all the tests
+	$(VENV_DIR)/bin/pytest tests --mpl --cov -rfsxEX --cov-report term-missing
+
+.PHONY: docs
+docs: $(VENV_DIR)  ## make the docs
+	cd doc; make html
 
-venv:
-	[ -d ./venv ] || python3 -m venv venv
-	./venv/bin/pip install --upgrade pip
-	./venv/bin/pip install -e .[tests,docs,deploy]
-	touch venv
+.PHONY: virtual-environment
+virtual-environment: $(VENV_DIR)  ## make virtual environment for development
+$(VENV_DIR): setup.py ci/environment-conda-default.txt ci/environment-conda-forge.txt
```

Comment:
My Make-foo is a bit rusty. I am not sure what this line is doing, but its removal leads to successfully installing an env.